pykechain/client.py

```diff
@@ -3047,3 +3048,77 @@ def active_banner(self) -> Banner:
             data = active_banner_list[0]
 
         return Banner(json=data, client=self)
+
+    def expiring_download(self, *args, **kwargs) -> ExpiringDownload:
+        """Retrieve a single expiring download.
+
+        :return: a single :class:`models.ExpiringDownload`
+        :raises NotFoundError: When no `ExpiringDownload` is found
+        :raises MultipleFoundError: When more than a single `ExpiringDownload` is found
+        """
+        return self._retrieve_singular(self.expiring_downloads, **args, **kwargs)
+
+    def expiring_downloads(
+            self,
+            pk: Optional[Text] = None,
+            expires_at: Optional[datetime.datetime] = None,
+            expires_in: Optional[int] = None,
+            **kwargs
+    ) -> List[ExpiringDownload]:
+        """Search fore xpiring downloads with optional pk, .
+
+        :param pk: if provided, filter the search for an expiring download by download_id
+        :type pk: basestring or None
+        :param expires_at: if provided, filter the search for the expires_at date and time
+        :type expires_at: datetime.datetime
+        :param expires_in: if provided, filter the search for the expires_in (in seconds)
+        :type expires_in: int
+        :return: list of Expiring Downloads objects
+        """
+        request_params = {
+            'id': check_uuid(pk)
+            # 'expires_in':
+        }
+        request_params.update(API_EXTRA_PARAMS['expiring_downloads'])
+
+        if kwargs:
+            request_params.update(**kwargs)
+
+        response = self._request('GET', self._build_url('expiring_downloads'), params=request_params)
+
+        if response.status_code != requests.codes.ok:  # pragma: no cover
+            raise NotFoundError("Could not retrieve Expiring Downloads", response=response)
+
+        return [ExpiringDownload(json=download, client=self) for download in response.json()['results']]
+
+    def create_expiring_download(
+            self,
+            expires_at: Optional[datetime.datetime] = None,
+            expires_in: Optional[int] = None,
+            content_path: Optional[Text] = None
+    ) -> ExpiringDownload:
+        """
+        Create an new Expiring Download.
+
+        :param expires_at:
+        :param expires_in:
+        :param content_path:
+        :return:
+        """
+        if isinstance(expires_at, datetime.datetime):
```

Comment:
dont we validate incoming arguments? I thought @JelleBoersma made some validation utilities for arguments.